Added Remotec ZTS-110 Thermostat

SkynetHome opened the pull request:
Added per Issue #2653

cdjackson commented on bundles/binding/org.openhab.binding.zwave/database/remotec/zts110.xml:
```diff
@@ -0,0 +1,88 @@
+			<Maximum>9999</Maximum>
+			<Size>1</Size>
+			<Label lang="en">Report Filter Counter</Label>
+			<Help lang="en"><![CDATA[READ ONLY: Data in hours]]> </Help>
```
This is a readonly parameter, so should have the ```<ReadOnly>true</ReadOnly>``` element added

cdjackson commented on bundles/binding/org.openhab.binding.zwave/database/remotec/zts110.xml:
```diff
@@ -0,0 +1,88 @@
+			
+			<Minimum>0</Minimum>
+			<Maximum>9999</Maximum>
+			<Size>1</Size>
```
Something doesn't look right - this has a range (min=0, max=9999) or 10000, but it's only 1 byte?

cdjackson commented on bundles/binding/org.openhab.binding.zwave/database/remotec/zts110.xml:
```diff
@@ -0,0 +1,88 @@
+			<Default>500</Default>
+			<Minimum>500</Minimum>
+			<Maximum>4000</Maximum>
+			<Size>1</Size>
```
Something doesn't look right - this has a range (min=500, max=4000) of 3500, but it's only 1 byte?

SkynetHome replied:
Admittedly this is more than pushing my boundaries of understanding byte
size etc. The ranges I took straight from product documentation. I tried to
find an existing database file to guide me but there are no existing
thermostats with this many if any programmed configurations in the database.
On Jul 1, 2015 1:48 PM, "Chris Jackson" <notifications@github.com> wrote:

> In bundles/binding/org.openhab.binding.zwave/database/remotec/zts110.xml
> <https://github.com/openhab/openhab/pull/2838#discussion_r33710764>:
>
> > +			<Index>2</Index>
> > +			<Type>int</Type>
> > +			<Default>2</Default>
> > +			<Minimum>1</Minimum>
> > +			<Maximum>4</Maximum>
> > +			<Size>1</Size>
> > +			<Label lang="en">Differntial</Label>
> > +			<Help lang="en"><![CDATA[Set Differential value from 1F/.5C to 4F/2.0C Available settings: 1 - 4]]> </Help>
> > +		</Parameter>
> > +		<Parameter>
> > +			<Index>3</Index>
> > +			<Type>int</Type>
> > +			<Default>500</Default>
> > +			<Minimum>500</Minimum>
> > +			<Maximum>4000</Maximum>
> > +			<Size>1</Size>
>
> Something doesn't look right - this has a range (min=500, max=4000) of
> 3500, but it's only 1 byte?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openhab/openhab/pull/2838/files#r33710764>.
>

cdjackson replied:
we need to get this right though. The fact that we can’t find another thermostat doesn’t help anyway since other thermostats will be different with different configuration…

Looking at the manual, I would change these params to 2 bytes long

SkynetHome replied:
Happy to fix it. Do I make changes to the file and resubmit in same branch
or do I need to close this pull and make changes and then recommit in new
branch with new pull?
On Jul 1, 2015 2:00 PM, "Chris Jackson" <notifications@github.com> wrote:

> In bundles/binding/org.openhab.binding.zwave/database/remotec/zts110.xml
> <https://github.com/openhab/openhab/pull/2838#discussion_r33712102>:
>
> > +			<Index>2</Index>
> > +			<Type>int</Type>
> > +			<Default>2</Default>
> > +			<Minimum>1</Minimum>
> > +			<Maximum>4</Maximum>
> > +			<Size>1</Size>
> > +			<Label lang="en">Differntial</Label>
> > +			<Help lang="en"><![CDATA[Set Differential value from 1F/.5C to 4F/2.0C Available settings: 1 - 4]]> </Help>
> > +		</Parameter>
> > +		<Parameter>
> > +			<Index>3</Index>
> > +			<Type>int</Type>
> > +			<Default>500</Default>
> > +			<Minimum>500</Minimum>
> > +			<Maximum>4000</Maximum>
> > +			<Size>1</Size>
>
> we need to get this right though. The fact that we can’t find another
> thermostat doesn’t help anyway since other thermostats will be different
> with different configuration… Looking at the manual, I would change these
> params to 2 bytes long
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openhab/openhab/pull/2838/files#r33712102>.
>

cdjackson replied:
Just make the changes, commit them, and push. It will automatically update the PR on github.

cdjackson commented on bundles/binding/org.openhab.binding.zwave/database/remotec/zts110.xml:
```diff
@@ -0,0 +1,88 @@
+			<Maximum>1</Maximum>
+			<Label lang="en">Controller Updates</Label>
+			<SetToController>true</SetToController>
+		</Group>
```
The manual shows that there are 2 configuration groups.

SkynetHome commented:
Great will make changes now and repush.

SkynetHome commented:
Hmmm....went to Eclipse to make changes and it refreshed and the zts100.xml disappeared! Is this normal? Am I supposed to recreate it?

cdjackson commented:
You need to make sure you change back to the appropriate branch. Files are only visible in the branch that they are committed/merged into.

SkynetHome commented:
OH WOW! Well that explains a lot. I didnt realize that github and eclipse were connected in that way. I knew that changes in one affected the other but I didnt realize that changing the branch on github affected what I could see in Eclipse that answers about 5 questions I had in one! Making changes now!

cdjackson commented:
Git and Eclipse aren’t connected. Git simply keeps the file system in synch with the repository. So if you add, change, or delete a file in one branch, then change to another branch, you see the files as they are in that branch - simple :)

It’s not Eclipse that does this - it’s just that all the files on the file system have changed.

SkynetHome commented:
Yeah this is starting to come together. I am beginning to be able to
visualize the flow and understand how this works.

As for this device when you say the manual shows 2 config  groups are you
referring to the other two association groups mentioned on page 30  for
heat pump and compressor? I don't see anything else under config
parameters. Note I didn't write a couple of parameters like Upper Limit of
set point because I don't know how to code in both of those centigrade and
farhenheit  ranges.
On Jul 1, 2015 3:36 PM, "Chris Jackson" <notifications@github.com> wrote:

> Git and Eclipse aren’t connected. Git simply keeps the file system in
> synch with the repository. So if you add, change, or delete a file in one
> branch, then change to another branch, you see the files as they are in
> that branch - simple :)
>
> It’s not Eclipse that does this - it’s just that all the files on the file
> system have changed.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openhab/openhab/pull/2838#issuecomment-117818012>.
>

SkynetHome commented:
On another note I think  I just discovered why the devices that I actually own and added are not letting me configure them and I think I have made the same mistake here so I would like to fix it all at once. 
On the configuration parameters I dont have an item and value assigned like this...

<Item>
     <value>0</value>
</item>

Instead its just
<Index>
<Type>
<Default>
etc 
etc
all in one column with no item and value tabbed in

So the configuration tab pops up but you cant get into it to set anything.
I suspect that I need to add that to my linear devices as well as all of the configuration parameters on this device as well!

SkynetHome commented:
Chris,

I dont know if you have seen this last commit, but it or I could use your help. I added everything that I thought I could add but there are a couple of parameters that I am not sure how to configure.

cdjackson commented:
I have - I’ve downloaded the manual again but I’m quite busy with work just at the moment :(

SkynetHome commented:
No worries, appreciate any help you can give. I think all I need to do is
add item and value settings for the parameters in question but thought I'd
wait until you have a look.
On Jul 2, 2015 1:04 PM, "Chris Jackson" <notifications@github.com> wrote:

> I have - I’ve downloaded the manual again but I’m quite busy with work
> just at the moment :(
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openhab/openhab/pull/2838#issuecomment-118112162>.
>

cdjackson commented on bundles/binding/org.openhab.binding.zwave/database/remotec/zts110.xml:
```diff
@@ -0,0 +1,143 @@
+			<Index>2</Index>
+			<Maximum>4</Maximum>
+			<Label lang="en">Compressor</Label>
+			<Help lang="en"><![CDATA[NOTE THIS PARAMETER NEEDS TO BE UPDATED I DONT KNOW THE PROPER WAY TO WRITE IT. ]]> </Help>
```
I'm not sure what you think the problem is - there are only 2 groups, so just have 2. Both should have the SetToController attribute.

cdjackson replied:
Sorry - I take it back - there are 3 - I was looking at the 100, not the 110 version.
Still, there's nothing to add to the associations.

SkynetHome replied:
What was throwing me was the table on page 31 showing different cases. I
didn't know if that had to be configured in the code as something the user
could define in the binding or if its just a representation of how the
device will work.
On Jul 2, 2015 2:38 PM, "Chris Jackson" <notifications@github.com> wrote:

> In bundles/binding/org.openhab.binding.zwave/database/remotec/zts110.xml
> <https://github.com/openhab/openhab/pull/2838#discussion_r33814668>:
>
> > +			<Label lang="en">Repeat Basic Set Counter</Label>
> > +			<Help lang="en"><![CDATA[NOTE THIS PARAMETER NEEDS TO BE UPDATED I DONT KNOW THE PROPER WAY TO WRITE IT. ]]> </Help>
> > +		</Parameter>
> > +	</Configuration>
> > +	<Associations>
> > +		<Group>
> > +			<Index>1</Index>
> > +			<Maximum>4</Maximum>
> > +			<Label lang="en">Heat Pump</Label>
> > +			<Help lang="en"><![CDATA[NOTE THIS PARAMETER NEEDS TO BE UPDATED I DONT KNOW THE PROPER WAY TO WRITE IT. ]]> </Help>
> > +		</Group>
> > +		<Group>
> > +			<Index>2</Index>
> > +			<Maximum>4</Maximum>
> > +			<Label lang="en">Compressor</Label>
> > +			<Help lang="en"><![CDATA[NOTE THIS PARAMETER NEEDS TO BE UPDATED I DONT KNOW THE PROPER WAY TO WRITE IT. ]]> </Help>
>
> Sorry - I take it back - there are 3 - I was looking at the 100, not the
> 110 version.
> Still, there's nothing to add to the associations.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openhab/openhab/pull/2838/files#r33814668>.
>

SkynetHome replied:
Just comited several more in my last push. Now have 14.
On Jul 2, 2015 2:40 PM, "Hydrilla Gorilla" <hydrillagorilla@gmail.com>
wrote:

> What was throwing me was the table on page 31 showing different cases. I
> didn't know if that had to be configured in the code as something the user
> could define in the binding or if its just a representation of how the
> device will work.
> On Jul 2, 2015 2:38 PM, "Chris Jackson" <notifications@github.com> wrote:
>
>> In bundles/binding/org.openhab.binding.zwave/database/remotec/zts110.xml
>> <https://github.com/openhab/openhab/pull/2838#discussion_r33814668>:
>>
>> > +			<Label lang="en">Repeat Basic Set Counter</Label>
>> > +			<Help lang="en"><![CDATA[NOTE THIS PARAMETER NEEDS TO BE UPDATED I DONT KNOW THE PROPER WAY TO WRITE IT. ]]> </Help>
>> > +		</Parameter>
>> > +	</Configuration>
>> > +	<Associations>
>> > +		<Group>
>> > +			<Index>1</Index>
>> > +			<Maximum>4</Maximum>
>> > +			<Label lang="en">Heat Pump</Label>
>> > +			<Help lang="en"><![CDATA[NOTE THIS PARAMETER NEEDS TO BE UPDATED I DONT KNOW THE PROPER WAY TO WRITE IT. ]]> </Help>
>> > +		</Group>
>> > +		<Group>
>> > +			<Index>2</Index>
>> > +			<Maximum>4</Maximum>
>> > +			<Label lang="en">Compressor</Label>
>> > +			<Help lang="en"><![CDATA[NOTE THIS PARAMETER NEEDS TO BE UPDATED I DONT KNOW THE PROPER WAY TO WRITE IT. ]]> </Help>
>>
>> Sorry - I take it back - there are 3 - I was looking at the 100, not the
>> 110 version.
>> Still, there's nothing to add to the associations.
>>
>> —
>> Reply to this email directly or view it on GitHub
>> <https://github.com/openhab/openhab/pull/2838/files#r33814668>.
>>
>

cdjackson replied:
I agree the table looks like it's omething that should go in a list. But in reality I think that doesn't work since there it's a time, but just a couple of specific options. I would just write all this in the help...

SkynetHome replied:
Will do, I'll finish up all the suggested changes in just a bit.
On Jul 2, 2015 2:46 PM, "Chris Jackson" <notifications@github.com> wrote:

> In bundles/binding/org.openhab.binding.zwave/database/remotec/zts110.xml
> <https://github.com/openhab/openhab/pull/2838#discussion_r33815385>:
>
> > +			<Label lang="en">Repeat Basic Set Counter</Label>
> > +			<Help lang="en"><![CDATA[NOTE THIS PARAMETER NEEDS TO BE UPDATED I DONT KNOW THE PROPER WAY TO WRITE IT. ]]> </Help>
> > +		</Parameter>
> > +	</Configuration>
> > +	<Associations>
> > +		<Group>
> > +			<Index>1</Index>
> > +			<Maximum>4</Maximum>
> > +			<Label lang="en">Heat Pump</Label>
> > +			<Help lang="en"><![CDATA[NOTE THIS PARAMETER NEEDS TO BE UPDATED I DONT KNOW THE PROPER WAY TO WRITE IT. ]]> </Help>
> > +		</Group>
> > +		<Group>
> > +			<Index>2</Index>
> > +			<Maximum>4</Maximum>
> > +			<Label lang="en">Compressor</Label>
> > +			<Help lang="en"><![CDATA[NOTE THIS PARAMETER NEEDS TO BE UPDATED I DONT KNOW THE PROPER WAY TO WRITE IT. ]]> </Help>
>
> I agree the table looks like it's omething that should go in a list. But
> in reality I think that doesn't work since there it's a time, but just a
> couple of specific options. I would just write all this in the help...
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/openhab/openhab/pull/2838/files#r33815385>.
>

cdjackson commented on bundles/binding/org.openhab.binding.zwave/database/remotec/zts110.xml:
```diff
@@ -0,0 +1,143 @@
+			<Label lang="en">Repeat Basic Set Counter</Label>
+			<Help lang="en"><![CDATA[NOTE THIS PARAMETER NEEDS TO BE UPDATED I DONT KNOW THE PROPER WAY TO WRITE IT. ]]> </Help>
+		</Parameter>
+	</Configuration>
```
There should be 13 parameters I think, so you're missing a few?

cdjackson commented on bundles/binding/org.openhab.binding.zwave/database/remotec/zts110.xml:
```diff
@@ -0,0 +1,143 @@
+			<Help lang="en"><![CDATA[Set Easy Mode 0: 24 Hours 1: 12 Hours DEFAULT: 12 Hours ]]> </Help>
+		</Parameter>
+		<Parameter>
+			<Index>10</Index>
```
I would just make this a 1 byte value the same as parameter 9.

cdjackson commented on bundles/binding/org.openhab.binding.zwave/database/remotec/zts110.xml:
```diff
@@ -0,0 +1,143 @@
+			<Minimum>0</Minimum>
+			<Maximum>1</Maximum>
+			<Size>1</Size>
+			<Label lang="en">Upper Limit of Setpoint A</Label>
```
I would just set the min and max to be the min/max alllowed of the C and F, so the smallest in C, and the largest in F. Same for both 6 and 7.

SkynetHome commented:
Ok sorry for the delay I had to install a new Dishwasher for the wife. I THINK I got this all sorted out. Added missing configuration parameters, made recommended changes and updated help text.

cdjackson commented on bundles/binding/org.openhab.binding.zwave/database/remotec/zts110.xml:
```diff
@@ -0,0 +1,184 @@
+			<Maximum>1</Maximum>
+			<Size>1</Size>
+			<Label lang="en">Scale of Temperature</Label>
+			<Help lang="en"><![CDATA[Set Temperature Scale to Farhenheit or Centigrade. DEFAULT =1 Farhenheit]]> </Help>
```
I didn't notice this previously, but this should be defined as a selection. So ```<Type>``` should be ```list``` and then you need to add two items -:
```xml
<Item>
    <Value>0</Value>
    <Label>Centigrade</Label>
</Item>
``` 
...and one for deg F as well (I've not checked that 0=degC for this device - this is just an example). The information on this is [here](https://github.com/cdjackson/HABmin2/wiki/Z-Wave-Product-Database#configuration-parameters)

SkynetHome commented:
Added the Items and values and changed a couple others that probably should be lists.

cdjackson commented:
Thanks.
